server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -108,7 +108,8 @@ final class FieldTypeLookup {
                             passThroughFieldAliases.put(name, conflict);
                             continue;
                         }
-                    } else if (fullNameToFieldType.containsKey(name)) {
+                    }
+                    if (fullNameToFieldType.containsKey(name)) {
```

Comment:
With this change, we won't track a new field that has higher priority than an existing one, i.e. conflict is not null.. 

Let me check the unittest too to understand the issue better.

Reply:
this should be fixed by b284a8c